apps/comps/hooks/useCompetitionBoosts.ts

```diff
@@ -0,0 +1,45 @@
+import { InfiniteData, useInfiniteQuery } from "@tanstack/react-query";
+import { UseInfiniteQueryResult } from "@tanstack/react-query";
+
+import { CompetitionBoostsResult } from "@recallnet/services";
+
+import { tanstackClient } from "@/rpc/clients/tanstack-query";
+import { RouterOutputs } from "@/rpc/router";
+import { getCompetitionPollingInterval } from "@/utils/competition-utils";
+
+/**
+ * Hook to fetch paginated boost allocations for a competition
+ * @param competitionId Competition ID
+ * @param limit Number of items per page
+ * @param enabled Whether the query is enabled
+ * @param competitionStatus Competition status for determining polling behavior
+ * @returns Infinite query result with boosts data and loadMore function
+ */
+export const useCompetitionBoosts = (
+  competitionId: string,
+  limit: number = 25,
+  enabled: boolean = true,
+  competitionStatus?: RouterOutputs["competitions"]["getById"]["status"],
+): UseInfiniteQueryResult<InfiniteData<CompetitionBoostsResult>, Error> => {
+  const baseOptions = tanstackClient.boost.competitionBoosts.queryOptions({
```

Comment:
> Check out the other uses of our rpc via the tanstack client. It simplifies all this and returns the query options you pass to useInfiniteQuery.

@asutula lmk if this is what you were thinking.

the `useInfiniteQuery` has a slightly different fn sig, so there's a bit of duplication, i guess. (i also dont _fully_ grok how to use it yet...tanstack & orpc are still a bit new to me.)

Reply:
There is a `tanstackClient.boost.competitionBoosts.infiniteQueryOptions`. That should remove the need to have `baseOptions` and all that additional complexity of extending them.

Reply:
ah idk how i missed that. okay, refactored to:
```ts
// ...
return useInfiniteQuery(
    tanstackClient.boost.competitionBoosts.infiniteOptions({
      input: enabled
        ? (pageParam: number) => ({
            competitionId,
            limit,
            offset: pageParam,
          })
        : skipToken,
      initialPageParam: 0,
      getNextPageParam: (lastPage) =>
        lastPage.pagination.hasMore
          ? lastPage.pagination.offset + lastPage.pagination.limit
          : undefined,
      staleTime: 60 * 1000,
      refetchInterval: () => getCompetitionPollingInterval(competitionStatus),
    }),
  );
```